image.go

```diff
@@ -175,48 +224,93 @@ func (i *containerImageRef) NewImageSource(sc *types.SystemContext, manifestType
 		if err != nil {
 			return nil, fmt.Errorf("error storing layer %q to file: %v", layerID, err)
 		}
-		layerDescriptor := v1.Descriptor{
-			MediaType: mediaType,
+		olayerDescriptor := v1.Descriptor{
+			MediaType: omediaType,
+			Digest:    destHasher.Digest(),
+			Size:      size,
+		}
+		omanifest.Layers = append(omanifest.Layers, olayerDescriptor)
+		dlayerDescriptor := docker.V2S2Descriptor{
+			MediaType: dmediaType,
 			Digest:    destHasher.Digest(),
 			Size:      size,
 		}
-		manifest.Layers = append(manifest.Layers, layerDescriptor)
+		dmanifest.Layers = append(dmanifest.Layers, dlayerDescriptor)
 		lastLayerDiffID := destHasher.Digest().String()
-		image.RootFS.DiffIDs = append(image.RootFS.DiffIDs, lastLayerDiffID)
+		oimage.RootFS.DiffIDs = append(oimage.RootFS.DiffIDs, lastLayerDiffID)
+		dimage.RootFS.DiffIDs = append(dimage.RootFS.DiffIDs, destHasher.Digest())
```

Comment:
The `DiffIDs` values should be digests of uncompressed data, AFAICS `destHasher` digests the possibly compressed stream.